test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextQueryTests.cs

```diff
@@ -0,0 +1,274 @@
+            ""OrgId"": 1,
+            ""Name"": ""Employee Two""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "Employees");
+
+            // Act
+            IQueryable<Employee> query = _defaultContext.Employees.Where(e => e.EmpNumber == 8);
+            Employee employee = query.First();
+
+            // Assert
+            Assert.Equal(expectedUri, query.ToString());
+            Assert.Equal(EmployeeType.PartTime, employee.EmpType);
+            Assert.Equal("Employee Two", employee.Name);
+        }
+
+        [Fact]
+        public void UseWhereToFilterByEnumKey_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string expectedUri = $"{ServiceRoot}/Employees?$filter=EmpType eq Microsoft.OData.Client.Tests.Tracking.EmployeeType'PartTime'";
+
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 8,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Employee 45""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "Employees");
+
+            // Act
+            var query = _defaultContext.Employees.Where(e => e.EmpType == EmployeeType.PartTime);
+            Employee employee = query.First();
+
+            // Assert
+            Assert.Equal(expectedUri, query.ToString());
+            Assert.Equal(8, employee.EmpNumber);
+            Assert.Equal(EmployeeType.PartTime, employee.EmpType);
+        }
+
+        [Fact]
+        public void UseByKeyToFilterByCompositeKeys_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string expectedUri = $"{ServiceRoot}/Employees(EmpNumber=8,EmpType=Microsoft.OData.Client.Tests.Tracking.EmployeeType'PartTime',OrgId=1)";
+
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 8,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Employee 24""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "Employees");
+
+            // Act
+            EmployeeSingle query = _defaultContext.Employees.ByKey(
+                new Dictionary<string, object>() { { "EmpNumber", 8 }, { "EmpType", EmployeeType.PartTime }, { "OrgId", 1 } });
+
+            Employee employee = query.GetValue();
+
+            // Assert
+            Assert.Equal(expectedUri, query.Query.ToString());
+            Assert.Equal("Employee 24", employee.Name);
+            Assert.Equal(EmployeeType.PartTime, employee.EmpType);
+        }
+
+        private void SetupContextWithRequestPipeline(DataServiceContext[] contexts, string response, string path)

```

Comment:
@WanjohiSammy What is the rationale behind make the first parameter an array? I don't see any scenario where you'd need to pass an array of `DataServiceContext`s

Reply:
@gathogojr 

That's correct John. To ensure code reusability, I reused the same logic and code from these tests:

- [DataServiceContextNoTrackingStreamsTests.cs#L321](https://github.com/OData/odata.net/blob/main/test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextNoTrackingStreamsTests.cs#L321)
- [ClientEntityTrackerTests.cs#L229](https://github.com/OData/odata.net/blob/main/test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/ClientEntityTrackerTests.cs#L229)

**For best practices, we can abstract this similar code in a helper method and reuse it.** Let me know if I can clean up this part of code in this PR.

Reply:
@gathogojr 

I have resolve this and also added a test to filter by enum key only using ByKey() in [commit 6a5667f](https://github.com/OData/odata.net/pull/3013/commits/6a5667f941b8fbe9d10929ef72909495775f2582)